banzai_nres/fitting.py

```diff
@@ -3,17 +3,21 @@
 from astropy.modeling import fitting, polynomial
 
 
-def fit_polynomial(y, error, mask=None, n_iter=5, x=None, sigma=5, order=3, domain=None):
+def fit_polynomial(y, error, mask=None, n_iter=5, x=None, sigma=5, order=3, domain=None, initial_weights=None):
     if mask is None:
         mask = np.zeros(y.shape, dtype=bool)
 
     if x is None:
         x = np.arange(len(y), dtype=float)
+
+    if initial_weights is None:
```

Comment:
This is now no longer used, revert to weights=error**-2 and get rid of the initial_weights feature